packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,151 @@
+  const severed = `SEVERED: ${iface.replace(/^Alleged: /, '')}`;
+  const thing = /** @type {any} */ (Far(severed, {}));
+  return thing;
+};
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * All purses, brands, etc. must be registered before serialization / unserialization.
+ * Neither the serializer nor the unserializer creates new presences.
+ */
+export const makeExportContext = () => {
+  const toVal = {
+    /** @type {MapStore<string, Purse>} */
+    purse: makeScalarMap(),
+  };
+  const fromVal = {
+    /** @type {MapStore<Purse, string>} */
+    purse: makeScalarMap(),
+  };
+  /** @type {MapStore<unknown, string>} */
+  const sharedData = makeScalarMap();
+
+  /**
+   * @param {string} slot
+   * @param {string} _iface
+   */
+  const slotToVal = (slot, _iface) => {
+    const kind = Object.keys(toVal).find(k => slot.startsWith(k));
+    assert(kind, X`bad slot kind: ${slot}`);
+    const id = slot.slice(kind.length + 1);
+    const val = toVal[kind].get(id); // or throw
+    return val;
+  };
+
+  let nonce = 0;
+
+  const valToSlot = val => {
+    if (sharedData.has(val)) {
+      return sharedData.get(val);
+    }
+    for (const kind of Object.keys(fromVal)) {
+      if (fromVal[kind].has(val)) {
+        const id = fromVal[kind].get(val);
+        return `${kind}:${id}`;
+      }
+    }
+    nonce += 1;
+    const slot = `unknown:${nonce}`;
+    sharedData.init(val, slot);
+    return slot;
+  };
+
+  return harden({
+    initPurseId: (id, purse) => {
+      toVal.purse.init(id, purse);
+      fromVal.purse.init(purse, id);
+    },
+    purseEntries: toVal.purse.entries,
+    // Public values.
+    initBoardId: (id, val) => {
+      sharedData.init(val, id);
+    },
+    ensureBoardId: (id, val) => {
+      if (sharedData.has(val)) return;
+      sharedData.init(val, id);
+    },
+    ...makeMarshal(valToSlot, slotToVal),
+  });
+};
+
+/**
+ * Make context for unserializing wallet or board data.
+ */
+export const makeImportContext = () => {
+  // constraint: none of these keys has another as a prefix.
+  // and none overlaps with board
+  const myData = {
+    /** @type {MapStore<string, Purse>} */
+    purse: makeScalarMap(),
+  };
+  /** @type {MapStore<string, unknown>} */
+  const sharedData = makeScalarMap();
+
+  const kindOf = slot => Object.keys(myData).find(k => slot.startsWith(k));
+
+  const slotToVal = {
+    /**
+     * @param {string} slot
+     * @param {string} iface
+     */
+    fromBoard: (slot, iface) => {
+      if (sharedData.has(slot)) {
+        return sharedData.get(slot);
+      }
+      const kind = kindOf(slot);
+      if (kind) {
+        assert.fail(X`bad shared slot ${q(slot)}`);
+      }
+      const it = makePresence(slot, iface);
+
+      sharedData.init(slot, it);
+      return it;
+    },
+
+    /**
+     * @param {string} slot
+     * @param {string} iface
+     */
+    fromPart: (slot, iface) => {
```

Comment:
I get `fromBoard` but what's a "Part"?

Reply:
lack of inspiration.

at one point, I called it a partitioning marshaler cuz of the separation into purses, brands, contacts, etc.

maybe `fromPrivate` or `fromMyWallet` or...?

Reply:
`fromMyWallet` seems most informative. it has to come from "my wallet" or "the board"?